pulser-core/pulser/channels/channels.py

```diff
@@ -475,7 +497,10 @@ def __repr__(self) -> str:
         return self.name + config
 
     def _to_dict(self) -> dict[str, Any]:
-        return obj_to_dict(self, **asdict(self))
+        params = {
+            f.name: getattr(self, f.name) for f in fields(self) if f.init
+        }
+        return obj_to_dict(self, _module="pulser.channels", **params)
```

Comment:
Cannot you use the `asdict` method provided by the `dataclass` library?

Reply:
You mean as a replacement for `obj_to_dict` or as it was being used before? I'll answer both, just in case.
- `obj_to_dict` is what is used by the JSON serializer and it includes some extra metadata that `asdict` won't.
- `asdict` will include all fields, even those that are not on `__init__`. This was then causing problems deserializing because `reusable_channels` is not in the `Device.__init__()`, so now we filter the fields accordingly.

Reply:
Got it, I understand now.